module/epochs/epoch_lookup_test.go

```diff
@@ -170,44 +160,49 @@ func (suite *EpochLookupSuite) TestEpochForViewWithFallback_CurrNextPrev() {
 // EpochLookup with an initial state of epoch fallback triggered.
 func (suite *EpochLookupSuite) TestEpochForViewWithFallback_EpochFallbackTriggered() {
 	epochs := []epochRange{suite.prevEpoch, suite.currEpoch, suite.nextEpoch}
-	suite.WithLock(func() {
-		suite.epochFallbackTriggered = true
-	})
 	suite.CommitEpochs(epochs...)
 	suite.CreateAndStartEpochLookup()
 	testEpochForViewWithFallback(suite.T(), suite.lookup, suite.state, epochs...)
 }
 
-// TestProtocolEvents_EpochFallbackTriggered tests constructing and subsequently querying
-// EpochLookup, where there is no epoch fallback at construction time,
-// but an epoch fallback happens later via an epoch event.
-func (suite *EpochLookupSuite) TestProtocolEvents_EpochFallbackTriggered() {
-	// initially, only current epoch is committed
-	suite.CommitEpochs(suite.currEpoch)
+// TestProtocolEvents_EpochExtended tests constructing and subsequently querying
+// EpochLookup, where we process an EpochExtended event and expect the latest
+// epoch final view to be updated with the updated final view of the current epoch
+// in the protocol state.
+func (suite *EpochLookupSuite) TestProtocolEvents_EpochExtended() {
+	// previous and current epochs will be committed
+	suite.CommitEpochs(suite.prevEpoch)
+	epoch, extension := suite.epochFixtureWithExtension(suite.currEpoch.counter, suite.currEpoch.firstView, suite.currEpoch.finalView+1000)
+	suite.epochQuery.Add(epoch)
+
+	epochs := []epochRange{
+		suite.prevEpoch,
+		{counter: suite.currEpoch.counter, firstView: suite.currEpoch.firstView, finalView: extension.FinalView},
+	}
+
+	header := unittest.BlockHeaderFixture()
+	suite.state.On("AtHeight", header.Height).Return(suite.snapshot).Times(4)
+
 	suite.CreateAndStartEpochLookup()
 
-	// trigger epoch fallback
-	suite.WithLock(func() {
-		suite.epochFallbackTriggered = true
-	})
-	suite.lookup.EpochFallbackModeTriggered(0, nil)
+	suite.lookup.EpochExtended(suite.currEpoch.counter, header, extension)
 
 	// wait for the protocol event to be processed (async)
 	assert.Eventually(suite.T(), func() bool {
-		_, err := suite.lookup.EpochForViewWithFallback(suite.currEpoch.finalView + 1)
+		_, err := suite.lookup.EpochForViewWithFallback(extension.FinalView)
 		return err == nil
 	}, 5*time.Second, 50*time.Millisecond)
 
 	// validate queries are answered correctly
-	testEpochForViewWithFallback(suite.T(), suite.lookup, suite.state, suite.currEpoch)
+	testEpochForViewWithFallback(suite.T(), suite.lookup, suite.state, epochs...)
 
 	// should handle multiple deliveries of the protocol event
-	suite.lookup.EpochFallbackModeTriggered(0, nil)
-	suite.lookup.EpochFallbackModeTriggered(0, nil)
-	suite.lookup.EpochFallbackModeTriggered(0, nil)
+	suite.lookup.EpochExtended(suite.currEpoch.counter, header, extension)
+	suite.lookup.EpochExtended(suite.currEpoch.counter, header, extension)
+	suite.lookup.EpochExtended(suite.currEpoch.counter, header, extension)
 
```

Comment:
For completeness, let's add a check to wait for the channel to drain:
```suggestion
	require.Eventually(bs.T(), func() bool {
		return len(bs.ctl.epochEvents) == 0
	}, time.Second, time.Millisecond)
```

(Since the channel capacity is 20, I think the current implementation is potentially race-y as there is no guarantee we actually process the events before the test continues. Not a problem with this PR.)